crates/sui-graphql-rpc/src/types/event/mod.rs

```diff
@@ -158,64 +115,67 @@ impl Event {
-                    // `tx_senders`.
-                    if let Some(sender) = &filter.sender {
-                        query = query.filter(
-                            events::dsl::tx_sequence_number.eq_any(
-                                tx_senders::dsl::tx_senders
-                                    .select(tx_senders::dsl::tx_sequence_number)
-                                    .filter(tx_senders::dsl::sender.eq(sender.into_vec())),
-                            ),
-                        )
-                    }
-
-                    if let Some(digest) = &filter.transaction_digest {
-                        // Since the event filter takes in a single tx_digest, we know that
-                        // there will only be one corresponding transaction. We can use
-                        // single_value() to tell the query planner that we expect only one
-                        // instead of a range of values, which will subsequently speed up query
-                        // execution time.
-                        query = query.filter(
-                            events::dsl::tx_sequence_number.nullable().eq(
-                                transactions::dsl::transactions
-                                    .select(transactions::dsl::tx_sequence_number)
-                                    .filter(
-                                        transactions::dsl::transaction_digest.eq(digest.to_vec()),
-                                    )
-                                    .single_value(),
-                            ),
-                        )
-                    }
-
-                    if let Some(module) = &filter.emitting_module {
-                        query = module.apply(query, events::dsl::package, events::dsl::module);
-                    }
-
-                    if let Some(type_) = &filter.event_type {
-                        query = type_.apply(
-                            query,
-                            events::dsl::event_type,
-                            events::dsl::event_type_package,
-                            events::dsl::event_type_module,
-                            events::dsl::event_type_name,
-                        );
-                    }
-
-                    query
-                })
+                let tx_hi: i64 = conn.first(move || {
+                    dsl::checkpoints.select(dsl::network_total_transactions)
+                        .filter(dsl::sequence_number.eq(checkpoint_viewed_at as i64))
+                })?;
+
+                query = add_bounds(query, &filter, &page, (tx_hi - 1) as u64);
+
+                let (prev, next, results) =
+                    page.paginate_raw_query::<EvLookup>(conn, checkpoint_viewed_at, query)?;
+
+                let ev_lookups = results
+                    .into_iter()
+                    .map(|x| (x.tx, x.ev))
+                    .collect::<Vec<(i64, i64)>>();
+
```

Comment:
(Although, do you even need to go through this step?)

Reply:
i haven't figured out how another way to move `results: impl Iterator` directly into the `conn.results` closure - 
i cant move `results` directly into `conn.results` because `impl Iterator` doesn't implement `Copy`